website/versioned_docs/version-20.x/troubleshooting/building-the-app.md

```diff
@@ -128,3 +128,27 @@ packagingOptions {
     exclude 'META-INF/LICENSE'
 }
 ```
+
+### Running Detox in a Rosetta environment
+
+When working with dependencies that require running your iOS app in a Rosetta simulator, you may encounter issues with the `detox build` command. These issues often relate to SwiftEmitModule or SwiftCompile errors. To resolve this, follow these steps:
+
+1. Modify your build command in the Detox configuration:
+
+```json
+"build": "xcodebuild -workspace ios/MyApp.xcworkspace -scheme MyApp -configuration Debug -sdk iphonesimulator -arch x86_64 -derivedDataPath ios/build"
+```
```

Comment:
Example is not enough, please emphasize that it's done by specifying `arch=x86_64`

Reply:
I think it will also work if you'll set that from the project's build settings

<img width="972" alt="image" src="https://github.com/user-attachments/assets/026be8cd-f0e5-4167-814a-c4e5a02f32c2">

Reply:
@asafkorem Setting it via the build settings might lead to conflicts with the necessary configuration for the app. In my project, for instance, I need specific build settings due to different dependencies. My goal is to adjust the architecture specifically for running the Detox tests, without altering the overall app build. That's why I chose to modify only the Detox build command. I want to emphasize that this is about changing Detox’s behavior, not my app’s – a crucial distinction!

Reply:
@differenzPriv True, I get your point. If your app can run in the Rosetta environment on Apple Silicon without Detox, then Detox is the only concern here, and your approach makes perfect sense. However, if that's not the case, it would be reasonable to add this architecture under the **Debug** configuration in Xcode. It might be worth mentioning this distinction in the doc IMO.

Reply:
In Xcode, you can enable "Show All Simulators". Here, we select the Rosetta simulator for the normal build. In the "Excluded Architectures" tab, we select the "arm64" architecture, not "x86_64". So, in my case, I only need to change the architecture for the "detox-build-command" and not the general one.

I like your suggestion to mention this distinction for better understanding and for choosing the right solution for a specific use case. 👍🏻

Reply:
Make sense @differenzPriv. 
@kagrawal98 can you add this note?

Reply:
@asafkorem have added the note in the end